src/c++11/impl.cc

```diff
@@ -81,6 +103,71 @@ std::unique_ptr<SpanImpl> TracerImpl::StartSpan(std::shared_ptr<TracerImpl> self
   return span;
 }
 
+bool TracerImpl::inject(SpanContext sc, const CarrierFormat& format, const CarrierWriter& opaque) {
+  const BuiltinCarrierFormat *bcf = dynamic_cast<const BuiltinCarrierFormat*>(&format);
+  if (bcf == nullptr) {
+    return false;
+  }
+  switch (bcf->type()) {
+  case BuiltinCarrierFormat::HTTPHeaders:
+  case BuiltinCarrierFormat::TextMap:
+    break;
+  default:
+    return false;
+  }
+
+  const TextMapWriter* carrier = dynamic_cast<const TextMapWriter*>(&opaque);
+  if (carrier == nullptr) {
+    return false;
+  }
+  carrier->Set(FieldNameTraceID, uint64ToHex(sc.trace_id()));
+  carrier->Set(FieldNameSpanID, uint64ToHex(sc.span_id()));
+  carrier->Set(FieldNameSampled, sc.sampled() ? "true" : "false");
+
+  sc.ForeachBaggageItem([carrier](const std::string& key,
+				  const std::string& value) {
+			  carrier->Set(std::string(PrefixBaggage) + key, value);
+			  return false;
+			});
+  return true;
+}
+
+SpanContext TracerImpl::extract(const CarrierFormat& format, const CarrierReader& opaque) {
+  const BuiltinCarrierFormat *bcf = dynamic_cast<const BuiltinCarrierFormat*>(&format);
+  if (bcf == nullptr) {
+    // TODO error handling
+    return SpanContext();
```

Comment:
Other langs return null when `extract` fails (... and make `ChildOf(null)` a noop). If that's a pain given the copy-don't-allocate approach you've taken to SpanContext, fine to just note this and move on.